### PR TITLE
add unit test case for pkg/api/resource_helpers.go

### PR DESCRIPTION
**Pull request by tanshanshan**
**What this PR does / why we need it**:

add unit test case for pkg/api/resource_helpers.go from #39559

Thanks.
**Special notes for your reviewer**:

**Release note**:

```release-note
```

### CODE REVIEW

File: pkg/api/resource_helpers_test.go [hunk 1]
```diff
@@ -118,3 +118,111 @@ func TestIsPodAvailable(t *testing.T) {
...
+				Status: ConditionTrue,
+				Type:   PodReady,
+			},
+			expected: false,
+		},
+	}
+
+	for i, test := range tests {
+		changedOrAadd := UpdatePodCondition(test.podStatus, test.podCondition)
+		if changedOrAadd != test.expected {
```
Comment (liggitt) [inline pkg/api/resource_helpers_test.go]: would probably be good to test the pod status looks as expected after this is called

Comment (tanshanshan) [inline pkg/api/resource_helpers_test.go]: `PodCondition.LastTransitionTime`  in  the function https://github.com/kubernetes/kubernetes/blob/master/pkg/api/resource_helpers.go#L147-L147 has been set current  local time . When pod condition has changed or has been added, this field can not be equal.
How to solve it making the actual value and expected value DeepEqual ?
Thanks.

Comment (liggitt) [inline pkg/api/resource_helpers_test.go]: You can verify the timestamp changed, then zero it out and compare the rest

**Comment by tanshanshan** [comment]
@liggitt  PTAL

**Comment by liggitt** [comment]
/lgtm /approve
@k8s-bot ok to test

**Comment by tanshanshan** [comment]
@liggitt  , I fix Jenkins verification error ,   PTAL.

**Pull request closed without merging**